Program.cs

```diff
@@ -1,57 +1,396 @@
+                        valid = true;
+                        break;
+                    case "exit":
+                        System.Environment.Exit(0);
+                        break;
+                    case "restart":
+                        break;
+                    default:
+                        Console.Write("\nInvalid selection, please key in a valid selection: ");
+                        break;
+                }
+            } while (!valid);
+            return citiesStats;
+        }
+        /// <summary>
+        /// Menu for province stats.
+        /// </summary>
+        /// <param name="cityStats"></param>
+        static void ProvinceMenu(Statistics cityStats)
+        {
+            bool valid = false;
+            string selection;
+            string titleText = "Provinces Menu";
+            string dash = new string('-', titleText.Length);
 
-                Console.Write("Select a data file to parse: ");
-                selection = Console.ReadKey().KeyChar;
-                if (selection >= '1' && selection <= '3')
-                    break;
-                Console.WriteLine("Invalid option. Valid options are 1, 2, 3.");
-            } while (true);
-            Console.WriteLine();
+            do
+            {
+                Console.WriteLine($"\n{dash}\n{titleText}\n{dash}\n");
+                Console.WriteLine("\nAvailable selections:");
+                Console.WriteLine("\t1) Display the population for a given province.");
+                Console.WriteLine("\t2) Display all the cities in a given province.");
+                Console.WriteLine("\t3) Rank provinces by population.");
+                Console.WriteLine("\t4) Rank provinces by the number of cities.");
+                Console.WriteLine("\t5) Display the capital city for the given province.");
+                Console.Write("Please make a selection(ex. 1, 2): ");
+                selection = Console.ReadLine();
+                string response;
+                switch (selection.ToLower())
+                {
+                    case "1":
+                        Console.Write("\nPlease key in the province name to see the population for that given province: ");
+                        do
+                        {
+                            response = Console.ReadLine();
+                            response = FormatAnswer(response);
+                            if (!cityStats.ProvinceIsValid(response))
+                            {
+                                Console.Write("\nInvalid response. Please pick from one of the following provinces and key in the province name: ");
+                                cityStats.DisplayProvinceList();
+                                Console.Write("Please key in one of the provinces listed above: ");
+                            }
+                        } while (!cityStats.ProvinceIsValid(response));
```

Comment:
I think the code on this switch statement would be a bit cleaner if you had a validateProvince method that took a prompt and returned a valid province string.